pkg/objectfile/cache.go

```diff
@@ -80,5 +81,14 @@ func fromProcess(pid int, m *profile.Mapping) (*MappedObjectFile, error) {
 }
 
 func cacheKey(pid int, m *profile.Mapping) string {
-	return path.Join("/proc", strconv.FormatInt(int64(pid), 10), m.BuildID)
+	b := make([]byte, 4*8)
+	// use all field needed in MappedObjectFile.computeBase to build a unique key
+	binary.BigEndian.PutUint64(b, m.Start)
+	binary.BigEndian.PutUint64(b, m.Limit)
+	binary.BigEndian.PutUint64(b, m.Offset)
+
+	// use pid and file to build unique identify
+	// do not use buildID, cause it tend to be larger strings
+	binary.BigEndian.PutUint64(b, uint64(pid))
+	return string(b) + m.File
```

Comment:
This causes a second allocation, let’s allocate only once when allocating the byte slice.

Reply:
how?

is this what you mean?
```
b = append(b, unsafeStringToSlice(m.File))
```